Feature: MoveIt!-Gazebo Integration

methylDragon opened the pull request:
I changed several packages in the entire Blue stack to work with MoveIt! Gazebo.

### Edited Packages
- blue_core
- blue_moveit
- blue_simulator

### Related PRs
- https://github.com/berkeleyopenarms/blue_simulator/pull/5#issue-281898034
- https://github.com/berkeleyopenarms/blue_moveit/pull/2#issue-281898515
- https://github.com/berkeleyopenarms/blue_core/pull/51#issue-281897805

### PR Specific Description
I added a world for simulating object pickup, as well as added a 3rd party mimic joint plugin for simulating mimic joints.

methylDragon committed:
Add mimic joint plugin

methylDragon committed:
Add pickup worlds

Normal world doesn't have gravity.

I also added a sky to make it look pretty.

methylDragon committed:
Add objects

methylDragon committed:
Add mimic joint params

methylDragon committed:
Modify launchfile to use gazebo-specific controller params

wuphilipp commented:
Thanks for your contribution! These changes look great! 

It seems you have added ```roboticsgroup_gazebo_plugins``` as a submodule. Is there a .gitmodules missing from the PR?

methylDragon committed:
Shift gazebo_controllers to blue_simulator repo

methylDragon committed:
Restore joint_torque_controller

methylDragon commented:
> Thanks for your contribution! These changes look great!
> 
> It seems you have added `roboticsgroup_gazebo_plugins` as a submodule. Is there a .gitmodules missing from the PR?

You're right about that. Let me fix it.

methylDragon committed:
Rename joint_trajectory_controller

methylDragon committed:
Add gitmodules file

methylDragon committed:
Alter mimic joint pid values

Note that they aren't used at the moment, but I'll leave this here in case someone wants to use them.

wuphilipp committed:
Move location of start controllers
